changelog/issue-6636.md

```diff
@@ -0,0 +1,12 @@
+audience: developers
+level: patch
+reference: issue 6636
+---
+Upgrades to modern Yarn version 4.
+
+To enable on your machine:
+
+```bash
+corepack enable
+yarn set version berry
```

Comment:
so many new words :) what does `berry` stands for?

Reply:
I can't find anything about those versions, can only see a branch in yarn's github project

btw, it is possible to add to `package.json`
```
{
  "packageManager": "yarn@3.1.0"
}
```

according to https://yarnpkg.com/blog/release/3.1#nodejs-corepack-integration 

but you still have to enable corepack before, and I'm not sure how node itself will enforce `yarn` usage

Reply:
> btw, it is possible to add to `package.json`

Added here :) https://github.com/taskcluster/taskcluster/pull/6646/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R10

> what does `berry` stands for?

It must just be their codename for their modern version of yarn, since it's been `berry` since v2 and now up to v4. I think it's just telling legacy yarn to upgrade.

I followed migration steps here https://yarnpkg.com/migration/guide#migration-steps.

Reply:
thanks! good to know